src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowsSettings/Helpers/ResultHelper.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+using Microsoft.CmdPal.Ext.WindowsSettings.Commands;
+using Microsoft.CmdPal.Ext.WindowsSettings.Helpers;
+using Microsoft.CmdPal.Ext.WindowsSettings.Properties;
+using Microsoft.CmdPal.Extensions.Helpers;
+
+namespace Microsoft.CmdPal.Ext.WindowsSettings;
+
+/// <summary>
+/// Helper class to easier work with List Items
+/// </summary>
+internal static class ResultHelper
+{
+    internal static List<ListItem> GetResultList(
+        in IEnumerable<Classes.WindowsSetting> list,
+        string query,
+        in string iconPath)
+    {
+        var resultList = new List<ListItem>(list.Count());
+
+        foreach (var entry in list)
+        {
+            var result = new ListItem(new OpenSettingsCommand(entry))
+            {
+                Icon = new(iconPath),
+                Subtitle = entry.JoinedFullSettingsPath,
+                Title = entry.Name,
+                MoreCommands = ContextMenuHelper.GetContextMenu(entry).ToArray(),
+            };
+
+            // TODO GH #126 investigate tooltips
+            // AddOptionalToolTip(entry, result);
+
+            // There is a case with MMC snap-ins where we don't have .msc files fort them. Then we need to show the note for this results in subtitle too.
+            // These results have mmc.exe as command and their note property is filled.
+            if (entry.Command == "mmc.exe" && !string.IsNullOrEmpty(entry.Note))
+            {
+                result.Subtitle += $"\u0020\u0020\u002D\u0020\u0020{Resources.Note}: {entry.Note}"; // "\u0020\u0020\u002D\u0020\u0020" = "<space><space><minus><space><space>"
+            }
+
+            // To not show duplicate entries we check the existing results on the list before adding the new entry. Example: Device Manager entry for Control Panel and Device Manager entry for MMC.
```

Comment:
frankly i'd do this _before_ you construct the `ListItem`s. May as well not bother doing you'll end up throwing out